README.md

```diff
@@ -42,87 +42,76 @@ Find a list of all the available options for your DogStatsD Client in the [DogSt
-# to close the instance is not necessary in this case since metrics are flushed on submission
-# but it is still a good practice and it explicitely closes the socket
+# ...
 statsd.close()
 ```
 
 or
 
 ```ruby
-# Import the library
-require 'datadog/statsd'
-
 # Create a DogStatsD client instance using UDS
 statsd = Datadog::Statsd.new(socket_path: '/path/to/socket/file', single_thread: true, buffer_max_pool_size: 1)
-...
-# to close the instance is not necessary in this case since metrics are flushed on submission
-# but it is still a good practice and it explicitely closes the socket
+# ...
 statsd.close()
 ```
 
 ### v5.x Common Pitfalls
 
-Version v5.x of `dogstatsd-ruby` is using a companion thread for preemptive flushing, it brings better performances for application having a high-throughput of statsd metrics, but it comes with new pitfalls:
+Version v5.x of `dogstatsd-ruby` is using a companion thread for flushing. This provides better performance, but you need to consider the following pitfalls:
 
-    * Applications forking after having created the dogstatsd instance: forking a process can't duplicate the existing threads, meaning that one of the processes won't have a companion thread to flush the metrics and will lead to missing metrics.
-    * Applications creating a lot of different instances of the client without closing them: it is important to close the instance to free the thread and the socket it is using or it will lead to thread leaks.
+1. Applications that use `fork` after having created the dogstatsd instance: the child process will need a new dogstatsd instance, as otherwise it will be unable to send metrics.
+
+2. Applications that create multiple instances of the client without closing them: it is important to `#close` the instance to free the thread and the socket it is using or it will lead to resources leaking (threads and sockets).
 
 If you are using [Sidekiq](https://github.com/mperham/sidekiq), please make sure to close the client instances that are instantiated. [See this example on using DogStatsD-ruby v5.x with Sidekiq](https://github.com/DataDog/dogstatsd-ruby/blob/master/examples/sidekiq_example.rb).
 
 If you are using [Puma](https://github.com/puma/puma) or [Unicorn](https://yhbt.net/unicorn.git), please make sure to create the instance of DogStatsD in the workers, not in the main process before it forks to create its workers. See [this comment for more details](https://github.com/DataDog/dogstatsd-ruby/issues/179#issuecomment-845570345).
 
-Applications that are in these situations but can't apply these recommendations should enable the `single_thread` mode which does not use a companion thread. Here is how to instantiate a client in this mode:
+Applications that run into issues but can't apply these recommendations should use the `single_thread` mode which avoids the use of the compainion thread.
+Here is how to instantiate a client in this mode:
 
 ```ruby
-# Import the library
-require 'datadog/statsd'
```

Comment:
Useful for irb

Reply:
See my question above about this

Reply:
I've no strong opinion on this, I'll merge your recommandation.